tools/building.py

```diff
@@ -453,6 +453,11 @@ def link_lld(args, target, external_symbol_list=None):
   for a in llvm_backend_args():
     cmd += ['-mllvm', a]
 
+  # Wasm exception handling. This is a CodeGen option for the LLVM backend, so
+  # wasm-ld needs to take this for the LTO mode.
+  if settings.EXCEPTION_HANDLING:
+    cmd += ['-mllvm', '-exception-model=wasm']
```

Comment:
I couldn't add this in [`llvm_backend_args`](https://github.com/emscripten-core/emscripten/blob/9bb622f6438c31e85890e2571499e0a3093e7f48/tools/building.py#L311-L334) because those flags are fed both to clang and lld, and apparently [LLVM CodeGen flags](https://github.com/llvm/llvm-project/blob/main/llvm/lib/CodeGen/CommandFlags.cpp) are not registered in clang; they work for opt, llc, and lld but not clang.

Reply:
I'm a little confused by this bit.. why want't it needed before?   

Is the exception model not embedded in the input IR files?

How are exceptions being enabled during compile if this flag is not passed to clang?

Reply:
During compile, users pass `-fwasm-exceptions` to clang, and it handles all the backend args. I'm not aware of a way that the exception model is explicitly encoded in the IR. There are of course exception-related instructions such as `invoke` in the IR, but they are interpreted differently based on the exception model.

Reply:
oh and to answer your first question more directly: strictly speaking this flag is only necessary when using LTO because clang handles it otherwise. I don't see this flag being conditioned on anything but I guess there's nothing in `settings` or whatever that tells us whether any of the input files are bitcode, so it probably does make sense to use it unconditionally.

Reply:
Thats is a shame that the input object don't encode the exception model :(   It means that mixing the model (e.g. compiling in one mode and linking in another) will fail in unexpected ways, right?

Anyway this change lgtm for now.